service/model/template.go

```diff
@@ -0,0 +1,48 @@
+package model
+
+import (
+	"context"
+
+	svcerr "release-manager/service/errors"
+
+	"github.com/google/uuid"
+)
+
+const (
+	releaseMsgTmplType = "release_message"
+)
+
+type TemplateRepository interface {
+	Insert(ctx context.Context, t Template) (Template, error)
+	ReadAll(ctx context.Context) ([]Template, error)
+	Read(ctx context.Context, id uuid.UUID) (Template, error)
+	Update(ctx context.Context, t Template) (Template, error)
+	Delete(ctx context.Context, id uuid.UUID) error
+}
+
+type Template struct {
+	ID         uuid.UUID
+	Type       TemplateType
+	ReleaseMsg ReleaseMessage
```

Comment:
The Template struct is currently directly linked to the concrete type ReleaseMessage. My initial goal was to abstract this relationship to simplify the process of adding new template types in the future. 

But it made the code more complex. In the end, I chose to keep things simple. It can be refactored if there will be need for new template type in the future.